### ISSUE TITLE
POST 500 not handled with error.response after CORS options returned 200

### ISSUE THREAD

**Issue by oonisim** [axios/axios#1960]
#### Summary

The way [Axios error handling ](https://github.com/axios/axios#handling-errors) provides to handle the POST to cross origin (React app development server to AWS API gateway) does not work as expected.

```
        axios.post(
            config.apiGateway.URL + ".../signin",
            payload
        ).then(response => {
            console.log(response)
            if(response.status == 200){
                console.log("Login successfull");
                this.setState({token: response.data.JWT_TOKEN});
            } else {
                console.log(response.status)
                this.setError()
            }
        }).catch(error => {
            this.setError()
            console.log(error)
            if (error.response) {
                console.log("--------------------------------------------------")
                // The request was made and the server responded with a status code
                // that falls out of the range of 2xx
                console.log(error.response.data);
                console.log(error.response.status);
                console.log(error.response.headers);
            } else if (error.request) {
                console.log("*************************")

                // The request was made but no response was received
                // `error.request` is an instance of XMLHttpRequest in the browser and an instance of
                // http.ClientRequest in node.js
                console.log(error.request);
            } else {
                console.log("++++++++++++++++++++++++")
                // Something happened in setting up the request that triggered an Error
                console.log('Error', error.message);
            }
            console.log(error.config);
        })
```
OPTIONS method returns 200 as below.
```
access-control-allow-headers: Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token
access-control-allow-methods: GET,OPTIONS,POST,PUT
access-control-allow-origin: *
content-length: 0
content-type: application/json
date: Wed, 16 Jan 2019 01:32:48 GMT
status: 200
via: 1.1 ca8c566e53d4556421f22afbb9802d0c.cloudfront.net (CloudFront)
x-amz-apigw-id: Tkp2JEBmIAMFROA=
x-amz-cf-id: HXEEGh3TUcOckodVBKKoA1PVpg6h7UB5Ui3F1svTxpMYovB8Dt7rqQ==
x-amzn-requestid: a2144ccd-192e-11e9-9b6f-2dcad2a22d99
```

Then POST method returns 500 as expected as intentionally sending a wrong password to test failure.
```
Request Method: POST
Status Code: 500 
Remote Address: 54.230.245.164:443
Referrer Policy: no-referrer-when-downgrade
```

#### Expected
error.response gets populated and can get error information from the error.response.

```
            if (error.response) {
                console.log("--------------------------------------------------")
                // The request was made and the server responded with a status code
                // that falls out of the range of 2xx
                console.log(error.response.data);
                console.log(error.response.status);
                console.log(error.response.headers);
            } else if (error.request) {
```

#### Actual

Get into "The request was made but no response was received" with " No 'Access-Control-Allow-Origin' header is present on the requested resource.". However "access-control-allow-origin: *" has been returned from OPTIONS.

```
Access to XMLHttpRequest at 'https://***-api.us-east-1.amazonaws.com/dev/***/signin' from origin 'http://localhost:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
Login.js:73 Error: Network Error
    at createError (createError.js:17)
    at XMLHttpRequest.handleError (xhr.js:87)
Login.js:82 *************************
Login.js:87 XMLHttpRequest {onreadystatechange: ƒ, readyState: 4, timeout: 0, withCredentials: false, upload: XMLHttpRequestUpload, …}
```

#### Context
axios 0.18.0 with Google Chrome 70.0.3538.77 in Ubuntu 18 LTS.
```
{
  "dependencies": {
    "axios": "^0.18.0",
    "react": "^16.7.0",
    "react-dom": "^16.7.0",
    "react-router-dom": "^4.3.1",
    "react-scripts": "2.1.3"
  },
  "browserslist": [
    ">0.2%",
    "not dead",
    "not ie <= 11",
    "not op_mini all"
  ]
}
```

#### Related

[Axios not returning response for POST on 500 error #1143](https://github.com/axios/axios/issues/1143) looks similar but the issue is due to OPTIONS method failing.

**Comment by oonisim** [comment]
My bad. 500 response did not provide the Access-Control-Allow-Origin header. Adding it fixed.
```
Login.js:74 Error: Request failed with status code 500
    at createError (createError.js:17)
    at settle (settle.js:19)
    at XMLHttpRequest.handleLoad (xhr.js:78)
Login.js:76 --------------------------------------------------
Login.js:79 
{message: "signin failed with [{'Message': u'Missing required…PASSWORD', 'Code': u'InvalidParameterException'}]"}
message: "signin failed with [{'Message': u'Missing required parameter PASSWORD', 'Code': u'InvalidParameterException'}]"
__proto__: Object
```

**Comment by oonisim** [comment]
closes #1960
